driver/bpf/filler_helpers.h

```diff
@@ -719,6 +719,8 @@ static __always_inline int __bpf_val_to_ring(struct filler_data *data,
 	unsigned int len_dyn = 0;
 	unsigned int len;
 
+	data->state->tail_ctx.curoff &= SCRATCH_SIZE_HALF;
+
```

Comment:
This makes the check below always succeed and feels suspicious. Without looking at the rest of the code, does this silently truncate something? If `tail_ctx.curoff == SCRATCH_SIZE_HALF + 2`, we'll chomp it to one byte.

I know we're fighting the BPF verifier here, but can we move this below the `>` check? If not, would a (mostly) harmless `& UINT_MAX` or `& (UINT_MAX >> 1)` (or maybe >> 2, just clearly larger than SCRATCH_SIZE_HALF) work here?

Or if we can't compare the values with `>` without masking out relevant bits, can we e.g.

```
unsigned long curoff = data->state->tail_ctx.curoff & SCRATCH_SIZE_HALF;
if(curoff != data->state->tail_ctx.curoff) { /* bad */ }
```

Or maybe `^` instead of `!=`? Basically, I don't like this silent masking :)

Reply:
I forgot to remove this one! It was a previous iteration, it's obviously wrong. Thanks for catching it!